Jellyfin.Api/Controllers/ScheduledTasksController.cs

```diff
@@ -0,0 +1,208 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Jellyfin.Api.Controllers
+{
+    /// <summary>
+    /// Scheduled Tasks Controller.
+    /// </summary>
+    [Authenticated]
+    public class ScheduledTasksController : BaseJellyfinApiController
+    {
+        private readonly ITaskManager _taskManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ScheduledTasksController"/> class.
+        /// </summary>
+        /// <param name="taskManager">Instance of the <see cref="ITaskManager"/> interface.</param>
+        public ScheduledTasksController(ITaskManager taskManager)
+        {
+            _taskManager = taskManager;
+        }
+
+        /// <summary>
+        /// Get tasks.
+        /// </summary>
+        /// <param name="isHidden">Optional filter tasks that are hidden, or not.</param>
+        /// <param name="isEnabled">Optional filter tasks that are enabled, or not.</param>
+        /// <returns>Task list.</returns>
+        [HttpGet]
+        [ProducesResponseType(typeof(TaskInfo[]), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
+        public IActionResult GetTasks(
+            [FromQuery] bool? isHidden = false,
+            [FromQuery] bool? isEnabled = false)
+        {
+            try
+            {
+                IEnumerable<IScheduledTaskWorker> tasks = _taskManager.ScheduledTasks.OrderBy(o => o.Name);
+
+                if (isHidden.HasValue)
+                {
+                    var hiddenValue = isHidden.Value;
+                    tasks = tasks.Where(o =>
+                    {
+                        var itemIsHidden = false;
+                        if (o.ScheduledTask is IConfigurableScheduledTask configurableScheduledTask)
+                        {
+                            itemIsHidden = configurableScheduledTask.IsHidden;
+                        }
+
+                        return itemIsHidden == hiddenValue;
+                    });
+                }
+
+                if (isEnabled.HasValue)
+                {
+                    var enabledValue = isEnabled.Value;
+                    tasks = tasks.Where(o =>
+                    {
+                        var itemIsEnabled = false;
+                        if (o.ScheduledTask is IConfigurableScheduledTask configurableScheduledTask)
+                        {
+                            itemIsEnabled = configurableScheduledTask.IsEnabled;
+                        }
+
+                        return itemIsEnabled == enabledValue;
+                    });
+                }
+
+                var taskInfos = tasks.Select(ScheduledTaskHelpers.GetTaskInfo);
+
+                return Ok(taskInfos);
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, e.Message);
```

Comment:
Exception middleware or MVC exception filter. I prefer the former though.

Reply:
I’ll create a PR for an exception middleware, sounds like something I should have learned about a while ago